src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +32,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        //this.value = trimmedAddress;
+        String[]SplitAddress=trimmedAddress.split(", ");
+        this.valueBlock = new Block(SplitAddress[0], isPrivate);
+        this.valueStreet = new Street(SplitAddress[1], isPrivate);
+        this.valueUnit = new Unit(SplitAddress[2], isPrivate);
+        this.valuePostalCode = new PostalCode(SplitAddress[3], isPrivate);
```

Comment:
Avoid the use of 'magic' numbers. Declare a variable for the position of each of these.